src/core/cookies/CookiesProvider.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { createContext, ReactNode, useContext } from 'react';
+
+import { getCookies, setCookie } from './cookies';
+
+import { CookieConsent } from "../../ui/cookies-consent";
+
+namespace CookiesProvider {
+  export interface State {
+    cookies: {[key: string]: string};
+    setCookie?: (key: string, value: string) => void
+  }
+
+  export interface Props {
+    children: ReactNode;
+  }
+}
+
+const STATE: CookiesProvider.State = {
+  cookies: getCookies()
+};
+
+const Context = createContext(STATE);
+
+class Provider extends React.Component<CookiesProvider.Props, typeof STATE> {
+  private _getCookies = () => {
```

Comment:
this method is not needed - create new methods / variables only if this saves you lines of code or increases code readability. This gives you nothing